src/common/log.cpp

```diff
@@ -580,13 +581,16 @@ wxLog *wxLog::SetThreadActiveTarget(wxLog *logger)
 {
     wxASSERT_MSG( !wxThread::IsMain(), "use SetActiveTarget() for main thread" );
 
-    wxLog * const oldLogger = wxThreadInfo.logger;
-    if ( oldLogger )
-        oldLogger->Flush();
+    if (wxThreadInfo) {
+        wxLog * const oldLogger = wxThreadInfo->logger;
+        if (oldLogger)
+            oldLogger->Flush();
 
-    wxThreadInfo.logger = logger;
+        wxThreadInfo->logger = logger;
+        return oldLogger;
+    }
 
-    return oldLogger;
+    wxASSERT_MSG( false, "SetThreadActiveTarget() only works when called from a wxThread" );
```

Comment:
This should be a `wxCHECK_MSG( wxThreadInfo, ...)` above (would also preserve the indent of the existing code).